keras/preprocessing/sequence.py

```diff
@@ -273,85 +272,200 @@ class TimeseriesGenerator(Sequence):
+    assert np.array_equal(y, np.array([[10], [11]]))
+
+    x, y = data_gen[-1]
+
+    assert np.array_equal(x, np.array([[[38], [40], [42], [44], [46]],
+                                   [[39], [41], [43], [45], [47]]]))
+    assert np.array_equal(y, np.array([[48], [49]]))
+
+    print "** test 2"
+    data_gen = TimeseriesGenerator(data, targets, length=10, batch_size=4)
+    assert len(data_gen) == 10
+    x, y = data_gen[0]
+    assert np.array_equal(x[1], np.array([[1], [2], [3], [4], [5], [6], [7], [8], [9], [10]]))
+    assert np.array_equal(y, np.array([[10], [11], [12], [13]]))
+
+    data_gen = TimeseriesGenerator(data, targets, length=10, reverse=True, batch_size=2)
+    x, y = data_gen[0]
+    assert np.array_equal(x[1,0], np.array([10]))
+
+    print "** test 3"
+    data_gen = TimeseriesGenerator(data, targets, length=10, sampling_rate=3, batch_size=2)
+
+    assert len(data_gen) == 10
+
+    print "** test 4 (stateful)"
+    data_gen = TimeseriesGenerator(data, targets, length=10, sampling_rate=2, batch_size=12, stateful=True)
+
+    assert data_gen.stride == 2
+    assert data_gen.batch_size == 10
+
+    print "** test 5 (text sequences seq2one)"
+    txt = bytearray("Keras is simple.")
+    data_gen = TimeseriesGenerator(txt, txt, length=10, batch_size=1)
+
+    for i in range(len(data_gen)):
+        print data_gen[i][0].tostring(), " -> '%s'" % data_gen[i][1].tostring()
+
+    assert data_gen[-1][0].shape == (1, 10) and data_gen[-1][1].shape==(1,)
+    assert data_gen[-1][0].tostring() == u" is simple"
+    assert data_gen[-1][1].tostring() == u"."
+
+    print "** test 6 (text sequences seq2seq)"
+    data_gen = TimeseriesGenerator(txt, txt, length=10, target_seq=True)
+
+    assert data_gen[-1][0].shape == (1, 10) and data_gen[-1][1].shape==(1,10,1)
+    for i in range(len(data_gen)):
+        print data_gen[i][0].tostring(), " -> '%s'" % data_gen[i][1].tostring()
+
+    assert data_gen[0][1].tostring() == u"eras is si"
+    
 
-    batch_0 = data_gen[0]
-    x, y = batch_0
-    assert np.array_equal(x,
-                          np.array([[[0], [2], [4], [6], [8]],
-                                    [[1], [3], [5], [7], [9]]]))
-    assert np.array_equal(y,
-                          np.array([[10], [11]]))
     ```
     """
 
     def __init__(self, data, targets, length,
                  sampling_rate=1,
                  stride=1,
-                 start_index=0,
-                 end_index=None,
-                 shuffle=False,
-                 reverse=False,
-                 batch_size=128):
-        self.data = data
-        self.targets = targets
+                 gap = None,
+                 start_index = 0,
+                 end_index = None,
+                 shuffle = False,
+                 reverse = False,
+                 target_seq = False,
+                 batch_size = 1,
+                 dtype = None,
+                 stateful = False):
+        
```

Comment:
nit: Extra blank line. Same at 390 & 430.

Reply:
Maybe should I put the keras_contrib version instead, because it is strictly backward compatible ?